src/server/scripts/Commands/cs_deserter.cpp

```diff
@@ -139,43 +172,85 @@ class deserter_commandscript : public CommandScript
     * .deserter bg remove
     * @endcode
     */
-    static bool HandleDeserterRemove(ChatHandler* handler, bool isInstance)
+    static bool HandleDeserterRemove(ChatHandler* handler, Optional<PlayerIdentifier> player, bool isInstance)
     {
-        Player* player = handler->getSelectedPlayer();
+        if (!player)
+        {
+            player = PlayerIdentifier::FromTargetOrSelf(handler);
+        }
+
         if (!player)
         {
             handler->SendSysMessage(LANG_NO_CHAR_SELECTED);
             handler->SetSentErrorMessage(true);
             return false;
         }
 
-        player->RemoveAura(isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+        Player* target = player->GetConnectedPlayer();
+
+        if (target)
+        {
+            target->RemoveAura(isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+            return true;
+        }
+
+        CharacterDatabase.Query("DELETE FROM character_aura WHERE guid = {} AND spell = {}", player->GetGUID().GetCounter(), isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+
+        return true;
+    }
+
+    static bool HandleDeserterRemoveAll(ChatHandler* handler, bool isInstance)
+    {
+        CharacterDatabase.Query("DELETE FROM character_aura WHERE spell = {} AND remainTime <= 1600000", isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+
+        std::shared_lock<std::shared_mutex> lock(*HashMapHolder<Player>::GetLock());
+        HashMapHolder<Player>::MapType const& onlinePlayerList = ObjectAccessor::GetPlayers();
+        for (HashMapHolder<Player>::MapType::const_iterator itr = onlinePlayerList.begin(); itr != onlinePlayerList.end(); ++itr)
+        {
+            Player* player = itr->second;
+            Aura* aura = player->GetAura(isInstance ? LFG_SPELL_DUNGEON_DESERTER : BG_SPELL_DESERTER);
+            if (aura && aura->GetDuration() <= 1600000)
```

Comment:
Can we please adjust this to cover the automated 30minutes deserter from RDF, too?
It should be `1800000`

Reply:
Oh damn. Thanks for pointing it out. I meant to do 30 minutes, so `1800000` but somehow messed up there :D